Description: Do not fail if the driver doesn't support NV_CTRL_NO_SCANOUT
 NV_CTRL_NO_SCANOUT is not supported by legacy drivers such as series
 173 and 96. Pretending that the property exists and is set to false
 prevents nvidia-settings from breaking compatibility with these drivers.
Forwarded: no
Bug: https://bugs.launchpad.net/ubuntu/+source/nvidia-settings/+bug/539196
Author: Alberto Milone <alberto.milone@canonical.com>

diff -Nurp nvidia-settings-195.36.08.orig/src/gtk+-2.x/ctkdisplayconfig-utils.c nvidia-settings-195.36.08/src/gtk+-2.x/ctkdisplayconfig-utils.c
--- nvidia-settings-195.36.08.orig/src/gtk+-2.x/ctkdisplayconfig-utils.c	2010-02-25 14:44:28.000000000 +0100
+++ nvidia-settings-195.36.08/src/gtk+-2.x/ctkdisplayconfig-utils.c	2010-04-06 19:47:14.335936746 +0200
@@ -2112,11 +2112,14 @@ static int gpu_add_screen_from_server(nv
     /* See if the screen is set to not scanout */
     ret = NvCtrlGetAttribute(screen->handle, NV_CTRL_NO_SCANOUT, &val);
     if (ret != NvCtrlSuccess) {
+        /*
         *err_str = g_strdup_printf("Failed to query NoScanout for "
                                    "screen %d.",
                                    screen_id);
         nv_warning_msg(*err_str);
         goto fail;
+        */
+        val = NV_CTRL_NO_SCANOUT_DISABLED;
     }
     screen->no_scanout = (val == NV_CTRL_NO_SCANOUT_ENABLED) ? TRUE : FALSE;
 
